shortlister/model.py

```diff
@@ -161,6 +210,58 @@ def clear_score(applicant: Applicant, criterion: Criterion):
         del applicant.scores[criterion]
 
 
+# text extraction
+
+
+def extract_info_from_text(cleaned_list):
+    """gets the section containing applicant information from extracted text"""
+
+    # fields names to get related applicant information
+    fields = (
+        "First Name",
+        "Last Name",
+        "Email Address",
+        "Preferred Phone Number",
+        "Postcode",
+        "Country & Region",
+    )
+    info = []
+
+    # removes header/footer and other irrelevant info
+    applicant_info = cleaned_list[1:-5]
+    right_to_work = cleaned_list[-5:-1]
+
+    # filter out the field name and retain only the info to applicant
+    for field in fields:
+        for i in applicant_info:
+            if field in i:
+                unclean = i.replace(field, "")
+                clean = re.sub(r"\s{2,}", " ", unclean)
+                info.append(clean.strip())
```

Comment:
We can make this a bit more straightforward. First, put the fields in a dictionary. We can use it to return the information in a more structured way than a list (with a list, you have to remember which position each bit of text is).

```
fields = dict.fromkeys(["First Name",
        "Last Name",
        "Email Address",
        "Preferred Phone Number",
        "Postcode",
        "Country & Region",], "<unretrievable>")
```

This creates a dictionary of fields and sets the default to the string "<unretrievable>". This is better than raising an error. Because, with an error, we get nothing - this way we can load something.

We can be more explicit when working with the string. Here's the pseudocode to implement:

```
# for each field to extract (i.e. loop over the dictionary keys)

    # for each line in text

        # if line starts with field - use `.startswith()` - `in` will search the whole line

            # remove the field from the line (use `.removeprefix()` method) 

            # strip whitespace from what's left - `.strip()` is sufficient, you don't need regex

            # put the captured text into the `fields` dictionary with the same key

            # continue with the outer loop - we don't need to check any more lines
```

Note, Python has an unusual construct to break out of nested loops:


```
for i in range(0,10):  # outer loop
    for j in range(10, 20):  # inner loop
        if some_condition:    # some condition is true
            break   # break out of this inner loop
    else:
        continue  # otherwise, continue this inner loop
````

Have a go, see how you get on.